packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,313 @@
+      "Sum of partner fractions must be less than or equal to 1."
+    );
+    emit CarbonOffsettingFundSet(partner, value);
+    return true;
+  }
+
+  /**
+   * @notice Calculates remaining CELO balance to mint.
+   * @return The remaining CELO balance to mint.
+   */
+  function getRemainingBalanceToMint() public view returns (uint256) {
+    return GOLD_SUPPLY_CAP - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @return The total balance minted by the MintGoldSchedule contract.
+   */
+  function getTotalMintedBySchedule() public view returns (uint256) {
+    return totalMintedBySchedule;
+  }
+
+  /**
+   * @return The currently mintable amount.
+   */
+  function getMintableAmount() public view returns (uint256) {
+    (uint256 targetGoldTotalSupply, , ) = getTargetGoldTotalSupply();
+    return targetGoldTotalSupply - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @notice Returns the target CELO supply according to the target schedule.
+   * @return targetGoldTotalSupply The target total CELO supply according to the target schedule.
+   * @return communityTargetRewards The community reward that can be minted according to the target schedule.
+   * @return carbonFundTargetRewards The carbon offsetting reward that can be minted according to the target schedule.
+   */
+  function getTargetGoldTotalSupply()
+    public
+    view
+    whenActivated
+    returns (
+      uint256 targetGoldTotalSupply,
+      uint256 communityTargetRewards,
+      uint256 carbonFundTargetRewards
+    )
+  {
+    require(block.timestamp > GENESIS_START_TIME, "GENESIS_START_TIME has not yet been reached.");
+    require(block.timestamp > l2StartTime, "l2StartTime has not yet been reached.");
+
+    uint256 timeSinceL2Start = block.timestamp - l2StartTime;
+    uint256 totalL2LinearSecondsAvailable = SECONDS_LINEAR - (l2StartTime - GENESIS_START_TIME);
+    uint256 mintedOnL1 = totalSupplyAtL2Start - GENESIS_GOLD_SUPPLY;
+
+    // Pay out half of all block rewards linearly.
+    uint256 l1LinearRewards = (GOLD_SUPPLY_CAP - GENESIS_GOLD_SUPPLY) / 2; //(200 million) includes validator rewards.
+
+    FixidityLib.Fraction memory l2LinearRewards = FixidityLib.newFixed(
+      l1LinearRewards - mintedOnL1
+    );
+
+    FixidityLib.Fraction memory linearRewardsToCommunity = l2LinearRewards.multiply(
+      communityRewardFraction
+    );
+
+    FixidityLib.Fraction memory linearRewardsToCarbon = l2LinearRewards.multiply(
+      carbonOffsettingFraction
+    );
+
+    bool isLinearDistribution = timeSinceL2Start < totalL2LinearSecondsAvailable;
+
+    if (isLinearDistribution) {
+      communityTargetRewards = (
+        linearRewardsToCommunity.multiply(FixidityLib.wrap(timeSinceL2Start)).divide(
+          FixidityLib.wrap(totalL2LinearSecondsAvailable)
+        )
+      ).fromFixed();
+
+      carbonFundTargetRewards = linearRewardsToCarbon
+        .multiply(FixidityLib.wrap(timeSinceL2Start))
```

Comment:
We are wrapping FixidityLib.wrap(timeSinceL2Start) and FixidityLib.wrap(totalL2LinearSecondsAvailable) multiple times - can we do it only once at the beginning and then you the variables ?